src/worker.cpp

```diff
@@ -877,11 +875,90 @@ void Worker::publishSystemVPD(const types::VPDMapVariant& parsedVpdMap)
     }
 }
 
+bool Worker::processPreAction(const std::string& i_vpdFilePath,
+                              const std::string& i_flagToProcess)
+{
+    if (i_vpdFilePath.empty() || i_flagToProcess.empty())
+    {
+        logging::logMessage(
+            "Invalid input parameter. Abort processing pre action");
+        return false;
+    }
+
+    if ((!utils::executePreAction(m_parsedJson, i_vpdFilePath,
+                                  i_flagToProcess)) &&
+        (i_flagToProcess.compare("collection") == constants::STR_CMP_SUCCESS))
+    {
+        // If the FRU is not there, clear the VINI/CCIN data.
+        // Enity manager probes for this keyword to look for this
+        // FRU, now if the data is persistent on BMC and FRU is
+        // removed this can lead to ambiguity. Hence clearing this
+        // Keyword if FRU is absent.
+        const auto& inventoryPath =
+            m_parsedJson["frus"][i_vpdFilePath].at(0).value("inventoryPath",
+                                                            "");
```

Comment:
Add a todo to delete FRU in case absent.
Deletion should be done just before priming the onject.
Do we need to delete object in case of any exception?

Reply:
Done.